### ISSUE TITLE
Switching rooms via back/forward browser buttons performs a slow room directory lookup

### ISSUE THREAD

**Issue by anoadragon453** [vector-im/riot-web#11079]
### Description

For a while now I've noticed that switching rooms via clicking on them via the sidebar is relatively fast, whereas using the back and forward buttons in my browser (done mainly through my mouse buttons), I'd have to wait 3-4 seconds before the timeline would appear.

I've realized this is now due to Riot making a [`GET /_matrix/client/r0/directory/room/{roomAlias}`](https://matrix.org/docs/spec/client_server/unstable#get-matrix-client-r0-directory-room-roomalias) request when it loads a room alias via URL. Because of this, this behaviour only occurs when switching to a room that has a default alias set (and thus the room URL is https://example.com/#/room/#bla:example.com instead of https://example.com/#/room/!123:example.com).

Instead of making this request every time, Riot should check if the room is already known, and immediately open it if so.

### Steps to reproduce

- Open a room that has a default alias set (you can see the room alias in the URL)
- Open another room
- Press back in your browser and notice Riot take a moment to load the room, as it's waiting on a directory lookup network request

Logs being sent: no

### Version information

- **Platform**: web
- **Browser**: Firefox
- **OS**: Debian buster
- **URL**: riot.im/develop

**Comment by t3chguy** [comment]
Duplicate of https://github.com/vector-im/riot-web/issues/10020

**Comment by anoadragon453** [comment]
Ah, thank you.
